Userland/Libraries/LibWeb/HTML/HTMLMediaElement.cpp

```diff
@@ -62,4 +94,544 @@ void HTMLMediaElement::pause() const
+        //            position to that time and, if jumped is still false, seek to that time.
+        // FIXME: 12. If there is no enabled audio track, then enable an audio track. This will cause a change event to be fired.
+
+        // 13. If there is no selected video track, then select a video track. This will cause a change event to be fired.
+        if (m_video_tracks->selected_index() == -1)
+            video_track->set_selected(true);
+
+        // FIXME: Once the readyState attribute reaches HAVE_CURRENT_DATA, after the loadeddata event has been fired, set the element's delaying-the-load-event
+        //        flag to false. This stops delaying the load event.
+    }
+
+    // -> Once the entire media resource has been fetched (but potentially before any of it has been decoded)
+    if (video_track != nullptr) {
+        // Fire an event named progress at the media element.
+        dispatch_event(TRY(DOM::Event::create(this->realm(), HTML::EventNames::progress)));
+
+        // Set the networkState to NETWORK_IDLE and fire an event named suspend at the media element.
+        m_network_state = NetworkState::Idle;
+        dispatch_event(TRY(DOM::Event::create(this->realm(), HTML::EventNames::suspend)));
+
+        // If the user agent ever discards any media data and then needs to resume the network activity to obtain it again, then it must queue a media
+        // element task given the media element to set the networkState to NETWORK_LOADING.
+    }
+
+    // FIXME: -> If the connection is interrupted after some media data has been received, causing the user agent to give up trying to fetch the resource
+    // FIXME: -> If the media data is corrupted
+    // FIXME: -> If the media data fetching process is aborted by the user
+    // FIXME: -> If the media data can be fetched but has non-fatal errors or uses, in part, codecs that are unsupported, preventing the user agent from
+    //           rendering the content completely correctly but not preventing playback altogether
+    // FIXME: -> If the media resource is found to declare a media-resource-specific text track that the user agent supports
+
+    return {};
+}
+
+// https://html.spec.whatwg.org/multipage/media.html#dedicated-media-source-failure-steps
+WebIDL::ExceptionOr<void> HTMLMediaElement::handle_media_source_failure()
+{
+    // FIXME: 1. Set the error attribute to the result of creating a MediaError with MEDIA_ERR_SRC_NOT_SUPPORTED.
+
+    // 2. Forget the media element's media-resource-specific tracks.
+    forget_media_resource_specific_tracks();
+
+    // 3. Set the element's networkState attribute to the NETWORK_NO_SOURCE value.
+    m_network_state = NetworkState::NoSource;
+
+    // FIXME: 4. Set the element's show poster flag to true.
+
+    // 5. Fire an event named error at the media element.
+    dispatch_event(TRY(DOM::Event::create(realm(), HTML::EventNames::error)));
```

Comment:
This might be impacted by https://github.com/SerenityOS/serenity/commit/4d87072201efc4ef9086588255fbb1bb993872ab